Sources/Sentry/SentryClient.m

```diff
@@ -392,6 +392,11 @@ - (SentryEvent *_Nullable)prepareEvent:(SentryEvent *)event
     // Need to do this after the scope is applied cause this sets the user if there is any
     [self setUserIdIfNoUserSet:event];
 
+    if (self.options.sendDefaultPii && nil == event.user.ipAddress) {
```

Comment:
M: should we check if `event.user` is null? otherwise buum

Reply:
Good point; I added a comment about that. The tests cover this edge case.